src/cnxml/de.ts

```diff
@@ -0,0 +1,679 @@
+        buildElement(editor, el, at, { type: 'docref', document }, INLINE)
+        normalizeLine(editor, at)
+    } else {
+        // TODO: record error
+        children(editor, el, at, INLINE)
+    }
+}
+
+function list(editor: Editor, el: Element, at: Path) {
+    const props = el.getAttribute('list-type') === 'enumerated'
+        ? {
+            type: 'list',
+            style: 'enumerated',
+            numberStyle: el.getAttribute('number-style') || 'arabic',
+            start: Number(el.getAttribute('start-value') || 1),
+        }
+        : {
+            type: 'list',
+            style: 'bulleted',
+            bullet: el.getAttribute('bullet') || 'bullet',
+        }
+
+    buildElement(editor, el, at, props, LIST)
+    normalizeBlock(editor, at)
+}
+
+const MARKS: { [key: string]: Omit<StyledText, 'text'> } = {
+    'sub': { position: 'subscript' },
+    'sup': { position: 'superscript' },
+    'bold': { strong: true },
+    'italics': { emphasis: true },
+}
+
+/** Deserialize a styling element */
+function mark(editor: Editor, el: Element, at: Path) {
+    const mark = el.tagName === 'emphasis'
+        ? el.getAttribute('effect') || 'bold'
+        : el.tagName
+    const props = MARKS[mark]
+    const end = Editor.pathRef(editor, at)
+
+    children(editor, el, at, INLINE)
+
+    if (props != null) {
+        Transforms.setNodes(editor, props, {
+            at: Editor.range(editor, at, Path.previous(end.current)),
+            match: Text.isText,
+        })
+    }
+
+    end.unref()
+}
+
+/** Deserialize media container */
+function media(editor: Editor, el: Element, at: Path) {
+    buildElement(editor, el, at, { type: 'media' }, MEDIA)
+
+    const node = Slate.Node.get(editor, at)
+
+    if (!node.children.some((n: Slate.Node) => n.type === 'media_alt')
+    && el.hasAttribute('alt')) {
+        const path = [...at, node.children.length]
+        editor.apply({
+            type: 'insert_node',
+            path,
+            node: {
+                type: 'media_alt',
+                children: [{ text: el.getAttribute('alt') }],
+            },
+        })
+        normalizeWhiteSpace(editor, path)
+    }
+
+    normalizeBlock(editor, at)
+}
+
+/** Deserialize a media item */
+function mediaItem(editor: Editor, el: Element, at: Path) {
+    const use = el.getAttribute('for')
+
+    buildElement(editor, el, at, {
+        type: 'media_' + el.tagName,
+        src: el.getAttribute('src'),
+        intendedUse: MediaUse.isMediaUse(use) ? use : 'all',
+    }, {})
+
+    normalizeVoid(editor, at)
+}
+
+/** Deserialize an admonition */
+function note(editor: Editor, el: Element, at: Path) {
+    buildElement(editor, el, at, {
+        type: 'admonition',
+        kind: el.getAttribute('type') || 'note',
+    }, { ...MIXED, title })
+    normalizeMixed(editor, at)
+}
+
+/** Deserialize <preformat> */
+function preformat(editor: Editor, el: Element, at: Path) {
+    buildElement(editor, el, at, { type: 'preformat' }, INLINE)
```

Comment:
Does this mean that there is no support for block `preformat`?

Reply:
No, as previously we only support block preformat, and convert all inline preformats to block ones.